src/http-util/src/lib.rs

```diff
@@ -210,3 +212,41 @@ pub async fn handle_tracing() -> impl IntoResponse {
         })),
     )
 }
+
+/// Close over a set of allowed CORS origins, and provide a helper method
+/// to check if an origin is allowed.
```

Comment:
Come to think of it, this whole thing should probably just be a function:

```rust
// TODO: docstring about how this handles subwildcards.
pub fn build_cors_allowed_origin<I>(allowed: I) -> AllowOrigin
where
    I: IntoIterator<Item = &'a HeaderValue>,
{
    let allowed = allowed.into_iter().cloned().collect();
    if allowed.iter().any(|o| o.as_bytes() == b'*') {
        AllowOrigin::any()
    } else {
        AllowOrigin::predicate(move |origin: &HeaderValue, _request_parts: _| {
             for val in &allowed {
                 // For each specified allow cors origin, check if it starts with
                 // a *. and allow anything from that glob. Otherwise check for
                 // an exact match.
                 if (val.as_bytes().starts_with(b"*.")
                     && origin.as_bytes().ends_with(&val.as_bytes()[1..]))
                     || origin == val
                 {
                     return true;
                 }
             }
             false
         })
    }
}
```